### DISCUSSION TITLE
Bindings via configuration file

### DISCUSSION THREAD

**Discussion by Mecanik** [cloudflare/next-on-pages#563] [Q&A]
Hi,

I am aware that currently Pages ignores `wrangler.toml`, however is there really nothing undocumented to add bindings? I hate the fact that I need to instruct users to go into the dashboard and add several bindings manually.

In this particular scenario I am cooking up something really cool for `next-on-pages` , but this issue is really holding me back. I saw some discussions about `.env` file and similar, but these do not have any effect on bindings.

Last chance would be to use the API and set the bindings; but even if I try to make a script for this, I would need the project details; something that I cannot get automatically if the user setup the project via the dashboard -> github.

Please let me know if there is something that can be done.

Many thanks

**Comment by Mecanik** [comment]
I forgot to mention that you can deploy this project via `wrangler`, however I did not see the bindings actually get attached from `wrangler.toml` AND if you deploy via `wrangler` **only the static assets get uploaded**. Something that is not really clarified on the [docs](https://developers.cloudflare.com/pages/framework-guides/deploy-a-nextjs-site/).

If someone wants to use next-on-pages with Edge Runtime, they must setup the project on the dashboard from Git, so that it builds there instead of your own environment.

**Comment by dario-piotrowicz** [comment]
Hi @Mecanik , as I mentioned in https://github.com/cloudflare/next-on-pages/issues/562 I am pretty sure that also the worker gets uploaded when you use `wrangler`, could you double check that? (is the `.vercel/output/static` directory name making you believe that it is only uploading static files by any chance?) 

Regarding setting the bindings, the team is looking into introducing a config file for Pages soon-ish, so you should be able to set bindings in that way, unfortunately there isn't an ETA for that 😓

In the meantime setting the bindings via the dashboard or via API calls is the only way to set bindings for Pages projects unfortunately 😢 

Regarding the rest API, the only things that you need for that are the account ID and the project same (see [api docs](https://developers.cloudflare.com/api/operations/pages-project-update-project)), I guess that's not something you can work with? 😓

**Reply by Mecanik** [reply]
Thanks, really sad there is no ETA. It's a pain for me and other users to set several binding and env variables manually via the dashboard. Using the API would not be a big deal, but I would need the project name and I have no idea how to get that. As mentioned, I would have liked to create several scripts that non-tech users can just execute and have everything done for them. I am 50% there lol.

**Reply by dario-piotrowicz** [reply]
yeah I'm very sorry @Mecanik 🙇  I am fully aware of the pain for users (myself included) of Pages not having a config file...

It is definitely planned and should happen soon-ish, I also can't wait to get that 😄 , but yeah for now there aren't much better alternatives that I am aware of 😢

**Reply by Mecanik** [reply]
It's what it is, thank you for the support. Appreciated.
